File: src/main/java/io/github/oleksivio/telegram/bot/core/controller/handler/check/impl/IntegerValidator.java
package io.github.oleksivio.telegram.bot.core.controller.handler.check.impl;

import io.github.oleksivio.telegram.bot.core.controller.handler.check.Validator;

public class IntegerValidator implements Validator<Integer> {
    private final int lessThan;
    private final int lessOrEqual;
    private final int moreThan;
    private final int moreOrEqual;

    public IntegerValidator(int lessThan, int lessOrEqual,
                            int moreThan, int moreOrEqual) {
        this.lessThan = lessThan;
        this.lessOrEqual = lessOrEqual;
        this.moreThan = moreThan;
        this.moreOrEqual = moreOrEqual;
    }

    @Override
    public boolean check(Integer argument) {
        return argument < lessThan && argument <= lessOrEqual && argument > moreThan && argument >= moreOrEqual;
    }

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/core/controller/handler/check/impl/IntegerValidator.kt
package io.github.oleksivio.telegram.bot.core.controller.handler.check.impl

import io.github.oleksivio.telegram.bot.core.controller.handler.check.Validator

class IntegerValidator(private val lessThan: Int, private val lessOrEqual: Int,
                       private val moreThan: Int, private val moreOrEqual: Int) : Validator<Int> {

    override fun invoke(checkedValue: Int): Boolean {
        return checkedValue < lessThan
                && checkedValue <= lessOrEqual
                && checkedValue > moreThan
                && checkedValue >= moreOrEqual
    }

}
